raiden/transfer/state_change.py

```diff
@@ -700,7 +711,8 @@ def __eq__(self, other):
             self.message_identifier == other.message_identifier and
             self.secret == other.secret and
             self.secrethash == other.secrethash and
-            self.balance_proof == other.balance_proof
+            self.balance_proof == other.balance_proof and
+            self.chain_id == other.chain_id
```

Comment:
chain_id is already part of the `BalanceProofSignedState`, it's not required here

Reply:
True. Nicely spotted. That's not even used anywhere right now. That's from before I realized putting `chain_id` in the `BalanceProofSignedState` was the way to go